lib/arel_extensions/visitors/mssql.rb

```diff
@@ -277,8 +277,17 @@ def visit_ArelExtensions_Nodes_Format o, collector
           end
           collector = visit o.left, collector
           if o.time_zone
```

Comment:
You don't need this if.

Actually, personally I would keep a single if for both opening and closing.  IMHO, it would be clearer.

```ruby
if o.time_zone
  collector << 'CONVERT(datetime'
  collector << LOADED_VISITOR::COMMA
  collector << ' '
  collector = visit o.left, collector
  case o.time_zone
   when Hash
     src_tz, dst_tz = o.time_zone.first.first, o.time_zone.first.second
     collector << ') AT TIME ZONE '
     collector = visit Arel::Nodes.build_quoted(src_tz), collector
     collector << ' AT TIME ZONE '
     collector = visit Arel::Nodes.build_quoted(dst_tz), collector
   when String
     collector << ') AT TIME ZONE '
     collector = visit Arel::Nodes.build_quoted(o.time_zone), collector
   end
else
  collector = visit o.left, collector
end

Reply:
And btw,
```
src_tz, dst_tz = o.time_zone.first
```

Reply:
Pattern matching will spoil me forever; adieu c++

Reply:
In C++ you do have that.

https://en.cppreference.com/w/cpp/language/structured_binding

Reply:
Ah C++17 ... nice.